filters/diag/diag.go

```diff
@@ -120,31 +118,15 @@ func (r *random) CreateFilter(args []interface{}) (filters.Filter, error) {
 	}
 
 	if l, ok := args[0].(float64); ok {
-		return &random{int(l)}, nil
+		return &random{reader: weakRandom(), len: int(l)}, nil
 	} else {
 		return nil, filters.ErrInvalidFilterParameters
 	}
 }
 
 func (r *random) ServeHTTP(w http.ResponseWriter, _ *http.Request) {
-	for n := 0; n < r.len; {
-		l := defaultChunkSize
-		if n+l > r.len {
-			l = r.len - n
-		}
-
-		b := make([]byte, l)
-		for i := 0; i < l; i++ {
-			b[i] = randomChars[rand.Intn(len(randomChars))]
-		}
-
-		ni, err := w.Write(b)
-		if err != nil {
-			log.Error("error while writing random content", err)
-			return
-		}
-
-		n += ni
+	if _, err := io.Copy(w, io.LimitReader(r.reader, int64(r.len))); err != nil {
```

Comment:
Is it better than
```go
func (r *random) Request(ctx filters.FilterContext) {
	ctx.Serve(&http.Response{
		StatusCode: http.StatusOK,
		Body:       ioutil.NopCloser(io.LimitReader(r.reader, r.len)),
	})
}
```

Reply:
No, your suggestion is better because it involves less moving parts. Thanks!